Dockerfile

```diff
@@ -94,22 +94,22 @@ RUN git clone -b @dop/redoc-cli@${REDOC_CLI_VERSION} --depth 1 https://github.co
 COPY --from=ts-compiler --chown=docsworker-xlarge /home/docsworker-xlarge/package*.json ./
 COPY --from=ts-compiler --chown=docsworker-xlarge /home/docsworker-xlarge/config config/
 COPY --from=ts-compiler --chown=docsworker-xlarge /home/docsworker-xlarge/build ./
-RUN npm install
+RUN npm install --legacy-peer-deps
 
 # Persistence module copy
 # Create directory and add permissions to allow node module installation
 RUN mkdir -p modules/persistence && chmod 755 modules/persistence
 COPY --from=ts-compiler --chown=docsworker-xlarge /home/docsworker-xlarge/modules/persistence/package*.json ./modules/persistence/
 COPY --from=ts-compiler --chown=docsworker-xlarge /home/docsworker-xlarge/modules/persistence/dist ./modules/persistence/
 ENV PERSISTENCE_MODULE_PATH=${WORK_DIRECTORY}/modules/persistence/index.js
-RUN cd ./modules/persistence/ && ls && npm install
+RUN cd ./modules/persistence/ && ls && npm install --legacy-peer-deps
 
 # OAS Page Builder module copy
 # Create directory and add permissions to allow node module installation
 RUN mkdir -p modules/oas-page-builder && chmod 755 modules/oas-page-builder
 COPY --from=ts-compiler --chown=docsworker-xlarge /home/docsworker-xlarge/modules/oas-page-builder/package*.json ./modules/oas-page-builder/
 COPY --from=ts-compiler --chown=docsworker-xlarge /home/docsworker-xlarge/modules/oas-page-builder/dist ./modules/oas-page-builder/
-RUN cd ./modules/oas-page-builder/ && npm install
+RUN cd ./modules/oas-page-builder/ && npm install --legacy-peer-deps
```

Comment:
Okay, so there's a really sad mess now with the serverless-esbuild plugin. It's not playing nice with a nested dependency of esbuild. Let me try to see if there's a different version that could remedy it.

Reply:
Reference: an old [failed build](https://github.com/mongodb/docs-worker-pool/actions/runs/5282909046/jobs/9558436067) with this spelled out

Reply:
Oh gotcha. Thanks for the context on this. Not sure about the best way to resolve this but if they're needed everywhere, it might be a good idea to leave a comment somewhere in the Dockerfile about why the flag is needed since it's only currently expected for the Snooty frontend.

(I can imagine ~~us~~ me wondering about the flag in the future)

Reply:
I think I figured it out!! Just going to build again on preprd to make sure the new downgrade is good. Should be! Thank you for pushing me on this!